velox/type/tests/TimestampConversionTest.cpp

```diff
@@ -132,76 +132,101 @@ TEST(DateTimeUtilTest, fromDateStrInvalid) {
+    EXPECT_EQ(1, castFromDateString("1970-1-02", mode));
 
-    EXPECT_EQ(1, castFromDateString("+1970-01-02", isIso8601));
-    EXPECT_EQ(-719893, castFromDateString("-1-1-1", isIso8601));
+    EXPECT_EQ(1, castFromDateString("+1970-01-02", mode));
+    EXPECT_EQ(-719893, castFromDateString("-1-1-1", mode));
 
-    EXPECT_EQ(0, castFromDateString(" 1970-01-01", isIso8601));
+    EXPECT_EQ(0, castFromDateString(" 1970-01-01", mode));
   }
 
-  EXPECT_EQ(3789391, castFromDateString("12345", false));
-  EXPECT_EQ(16436, castFromDateString("2015", false));
-  EXPECT_EQ(16495, castFromDateString("2015-03", false));
-  EXPECT_EQ(16512, castFromDateString("2015-03-18T", false));
-  EXPECT_EQ(16512, castFromDateString("2015-03-18T123123", false));
-  EXPECT_EQ(16512, castFromDateString("2015-03-18 123142", false));
-  EXPECT_EQ(16512, castFromDateString("2015-03-18 (BC)", false));
+  EXPECT_EQ(3789391, castFromDateString("12345", ParseMode::kNonStandardCast));
+  EXPECT_EQ(16436, castFromDateString("2015", ParseMode::kNonStandardCast));
+  EXPECT_EQ(16495, castFromDateString("2015-03", ParseMode::kNonStandardCast));
+  EXPECT_EQ(
+      16512, castFromDateString("2015-03-18T", ParseMode::kNonStandardCast));
+  EXPECT_EQ(
+      16512,
+      castFromDateString("2015-03-18T123123", ParseMode::kNonStandardCast));
+  EXPECT_EQ(
+      16512,
+      castFromDateString("2015-03-18 123142", ParseMode::kNonStandardCast));
+  EXPECT_EQ(
+      16512,
+      castFromDateString("2015-03-18 (BC)", ParseMode::kNonStandardCast));
 
-  EXPECT_EQ(0, castFromDateString("1970-01-01 ", false));
-  EXPECT_EQ(0, castFromDateString(" 1970-01-01 ", false));
+  EXPECT_EQ(0, castFromDateString("1970-01-01 ", ParseMode::kNonStandardCast));
+  EXPECT_EQ(0, castFromDateString(" 1970-01-01 ", ParseMode::kNonStandardCast));
 }
 
 TEST(DateTimeUtilTest, castFromDateStringInvalid) {
   auto testCastFromDateStringInvalid = [&](const StringView& str,
-                                           bool isIso8601) {
-    if (isIso8601) {
+                                           ParseMode mode) {
+    auto kStandardCastErr =
+        "Unable to parse date value: \"{}\"."
+        "Valid date string pattern is (YYYY-MM-DD), "
+        "and can be prefixed with [+-]";
+    auto kNonStandardCastErr =
+        "Unable to parse date value: \"{}\"."
+        "Valid date string patterns include "
+        "(yyyy*, yyyy*-[m]m, yyyy*-[m]m-[d]d, "
+        "yyyy*-[m]m-[d]d *, yyyy*-[m]m-[d]dT*), "
+        "and any pattern prefixed with [+-]";
+    auto kNonStandardNoTimeCastErr =
+        "Unable to parse date value: \"{}\"."
+        "Valid date string patterns include "
+        "(yyyy*, yyyy*-[m]m, yyyy*-[m]m-[d]d, "
+        "yyyy*-[m]m-[d]d *), "
+        "and any pattern prefixed with [+-]";
+    if (mode == ParseMode::kStandardCast) {
+      VELOX_ASSERT_THROW(
+          castFromDateString(str, mode),
+          fmt::format(kStandardCastErr, std::string(str.data(), str.size())));
```

Comment:
This doesn't work as fmt::format requires constant string or fmt::runtime. I'm fixing that.

Also, noticed there is no space between sentences. Fixing that too.

Reply:
Hmm, it seemed to be working when I ran the test locally. I think the error strings are declared as constants.